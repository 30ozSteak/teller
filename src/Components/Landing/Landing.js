import React, { Component } from "react";
import "./Landing.css";
import FavoritesContainer from "../FavoritesContainer/FavoritesContainer";
import LandingCurrencyContainer from "../LandingCurrencyContainer/LandingCurrencyContainer";
import NewsContainer from "../../Components/NewsContainer/NewsContainer";
import CurrencyExpanded from "../../Components/CurrencyExpanded/CurrencyExpanded"

class Landing extends Component {
  constructor(props) {
    super(props);

    const { favorites, addToFavorites, removeFromFavorites, abbrevCurrencies, setFilter, removeLoginState, token, } = this.props
    
    this.state = {
      active: false,
      news: [],
      displayedCurrency: "",
    };
  }

  toggleActive = () => {
    this.setState({
      active: !this.state.active
    });
  };

  displayExpanded = (name) => {
    if(!this.state.displayedCurrency) {
      this.setState({
        displayedCurrency: this.props.abbrevCurrencies[0].name
      })
    } else {
      this.setState({
        displayedCurrency: name
      })
    }
  }

  render() {
    const {
      favorites,
      addToFavorites,
      removeFromFavorites,
      abbrevCurrencies,
      setFilter,
      token
    } = this.props;

    return (
      <div className="landing-literal">
        <NewsContainer />
        <FavoritesContainer 
          favorites={favorites}
          removeFromFavorites={removeFromFavorites}/>
        <LandingCurrencyContainer 
          setFilter={setFilter}
          addToFavorites={addToFavorites}
          abbrevCurrencies={abbrevCurrencies}
          displayExpanded={this.displayExpanded}
          token={token}
        />
        {/*<CurrencyExpanded
                  currencies={abbrevCurrencies}
                  addToFavorites={addToFavorites}
                  displayedCurrency={this.state.displayedCurrency} />*/}
      </div>
    );
  }
}

export default Landing;
