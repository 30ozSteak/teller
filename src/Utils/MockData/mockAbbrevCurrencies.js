export const mockAbbrevCurrencies = [
  {
    "symbol": "BTC",
    "price": 3586.16,
    "percent_change": 0.83,
    "name": "Bitcoin",
    "rank": "1",
  },
  {
    "symbol": "XRP",
    "price": 0.33,
    "percent_change": 3.94,
    "name": "XRP",
    "rank": "2",
  },
  {
    "symbol": "ETH",
    "price": 95.07,
    "percent_change": 0.20,
    "name": "Ethereum",
    "rank": "3",
  },
  {
    "symbol": "EOS",
    "price": 2.54,
    "percent_change": 6.87,
    "name": "EOS",
    "rank": "4",
  },
  {
    "symbol": "XLM",
    "price": 0.11,
    "percent_change": -0.58,
    "name": "Stellar",
    "rank": "5",
  },
  {
    "symbol": "USDT",
    "price": 1.02,
    "percent_change": -0.37,
    "name": "Tether",
    "rank": "6",
  },
  {
    "symbol": "BCH",
    "price": 100.65,
    "percent_change": 12.32,
    "name": "Bitcoin Cash",
    "rank": "7",
  },
  {
    "symbol": "LTC",
    "price": 28.99,
    "percent_change": -1.97,
    "name": "Litecoin",
    "rank": "8",
  },
  {
    "symbol": "BSV",
    "price": 81.84,
    "percent_change": -1.04,
    "name": "Bitcoin SV",
    "rank": "9",
  },
  {
    "symbol": "TRX",
    "price": 0.01,
    "percent_change": 0.17,
    "name": "TRON",
    "rank": "10",
  }
]