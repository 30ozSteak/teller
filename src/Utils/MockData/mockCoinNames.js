export const mockCoinNames = [
  "Bitcoin",
  "XRP",
  "Ethereum",
  "EOS",
  "Stellar",
  "Tether",
  "Bitcoin Cash",
  "Litecoin",
  "Bitcoin SV",
  "TRON"
]