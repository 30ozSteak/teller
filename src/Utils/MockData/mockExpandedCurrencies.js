export const mockExpandedCurrencies = [
    {
        "symbol": "BTC",
        "price": 3586.16,
        "percent_change": 0.83,
        "name": "Bitcoin",
        "rank": "1",
        "supply": 17429837.00,
        "market_cap": 62506159630.72,
    },
    {
        "symbol": "XRP",
        "price": 0.33,
        "percent_change": 3.94,
        "name": "XRP",
        "rank": "2",
        "supply": 40762365544.00,
        "market_cap": 13612140064.75,
    },
    {
        "symbol": "ETH",
        "price": 95.07,
        "percent_change": 0.20,
        "name": "Ethereum",
        "rank": "3",
        "supply": 103875264.97,
        "market_cap": 9875418262.80,
    },
    {
        "symbol": "EOS",
        "price": 2.54,
        "percent_change": 6.87,
        "name": "EOS",
        "rank": "4",
        "supply": 906245117.60,
        "market_cap": 2299890068.43,
    },
    {
        "symbol": "XLM",
        "price": 0.11,
        "percent_change": -0.58,
        "name": "Stellar",
        "rank": "5",
        "supply": 19171055431.43,
        "market_cap": 2101330537.32,
    },
    {
        "symbol": "USDT",
        "price": 1.02,
        "percent_change": -0.37,
        "name": "Tether",
        "rank": "6",
        "supply": 1856421735.62,
        "market_cap": 1884620732.28,
    },
    {
        "symbol": "BCH",
        "price": 100.65,
        "percent_change": 12.32,
        "name": "Bitcoin Cash",
        "rank": "7",
        "supply": 17517750.00,
        "market_cap": 1763235900.72,
    },
    {
        "symbol": "LTC",
        "price": 28.99,
        "percent_change": -1.97,
        "name": "Litecoin",
        "rank": "8",
        "supply": 59624662.94,
        "market_cap": 1728272479.52,
    },
    {
        "symbol": "BSV",
        "price": 81.84,
        "percent_change": -1.04,
        "name": "Bitcoin SV",
        "rank": "9",
        "supply": 17517010.94,
        "market_cap": 1433656314.37,
    },
    {
        "symbol": "TRX",
        "price": 0.01,
        "percent_change": 0.17,
        "name": "TRON",
        "rank": "10",
        "supply": 66464300491.53,
        "market_cap": 951584289.18,
    }
]