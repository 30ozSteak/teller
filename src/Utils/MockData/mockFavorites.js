export const mockFavorites = [
  {name: "DopeCoin", price: 1000000, percent_change: 1.00},
  {name: "DoperCoin", price: 1000000, percent_change: 1.00},
  {name: "DopestCoin", price: 1000000, percent_change: 1.00},
  {name: "Even DoperCoin", price: 1000000, percent_change: 1.00},
  {name: "DopestCoinOfAll", price: 1000000, percent_change: 1.00},
]