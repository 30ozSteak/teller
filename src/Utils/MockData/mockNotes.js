export const mockNotes = [
  {
    title: "Buy some bitcoins!", 
    body: "I really need to invest in some bitcoins while the price is down. That way, once it goes back up, I'll be RICH!"
  },
  {
    title: "Ethereum",
    body: "The price of ethereum is going to drop next Tuesday! Make arrangements"
  },
  {
    title: "Never again!",
    body: "Never ever ever ever ever again will I invest in XRP. I almost lost everything!"
  }
]