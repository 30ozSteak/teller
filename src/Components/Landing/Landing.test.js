import React from 'react'
import { shallow } from 'enzyme'
import Landing from './Landing'
import { mockCurrencies } from '../../Utils/MockData/mockCurrencies'
import { mockAbbrevCurrencies } from '../../Utils/MockData/mockAbbrevCurrencies'

describe('Landing Component', () => {
  let wrapper
  let favorites = mockCurrencies.slice(0, 5)
  let mockAddToFavorites = jest.fn()
  let mockRemoveFromFavorites = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Landing
                        favorites={favorites}
                        addToFavorites={mockAddToFavorites}
                        removeFromFavorites={mockRemoveFromFavorites}
                        abbrevCurrencies={mockAbbrevCurrencies}
                     />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})