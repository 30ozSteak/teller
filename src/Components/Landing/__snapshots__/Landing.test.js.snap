// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    abbrevCurrencies={
      Array [
        Object {
          "name": "Bitcoin",
          "percent_change": 0.83,
          "price": 3586.16,
          "rank": "1",
          "symbol": "BTC",
        },
        Object {
          "name": "XRP",
          "percent_change": 3.94,
          "price": 0.33,
          "rank": "2",
          "symbol": "XRP",
        },
        Object {
          "name": "Ethereum",
          "percent_change": 0.2,
          "price": 95.07,
          "rank": "3",
          "symbol": "ETH",
        },
        Object {
          "name": "EOS",
          "percent_change": 6.87,
          "price": 2.54,
          "rank": "4",
          "symbol": "EOS",
        },
        Object {
          "name": "Stellar",
          "percent_change": -0.58,
          "price": 0.11,
          "rank": "5",
          "symbol": "XLM",
        },
        Object {
          "name": "Tether",
          "percent_change": -0.37,
          "price": 1.02,
          "rank": "6",
          "symbol": "USDT",
        },
        Object {
          "name": "Bitcoin Cash",
          "percent_change": 12.32,
          "price": 100.65,
          "rank": "7",
          "symbol": "BCH",
        },
        Object {
          "name": "Litecoin",
          "percent_change": -1.97,
          "price": 28.99,
          "rank": "8",
          "symbol": "LTC",
        },
        Object {
          "name": "Bitcoin SV",
          "percent_change": -1.04,
          "price": 81.84,
          "rank": "9",
          "symbol": "BSV",
        },
        Object {
          "name": "TRON",
          "percent_change": 0.17,
          "price": 0.01,
          "rank": "10",
          "symbol": "TRX",
        },
      ]
    }
    addToFavorites={[MockFunction]}
    favorites={
      Array [
        Object {
          "market_cap_usd": "62506159630.7222615731538755",
          "max_supply": "21000000.0000000000000000",
          "name": "Bitcoin",
          "percent_change_24_hr": "0.8320854828625978",
          "price_usd": "3586.1585871814097615",
          "rank": "1",
          "supply": "17429837.0000000000000000",
          "symbol": "BTC",
        },
        Object {
          "market_cap_usd": "13612140064.7471135264064432",
          "max_supply": "100000000000.0000000000000000",
          "name": "XRP",
          "percent_change_24_hr": "3.9433058589033265",
          "price_usd": "0.3339389135808078",
          "rank": "2",
          "supply": "40762365544.0000000000000000",
          "symbol": "XRP",
        },
        Object {
          "market_cap_usd": "9875418262.8035156239002594",
          "max_supply": null,
          "name": "Ethereum",
          "percent_change_24_hr": "0.2048869680495625",
          "price_usd": "95.0699694086438060",
          "rank": "3",
          "supply": "103875264.9678000000000000",
          "symbol": "ETH",
        },
        Object {
          "market_cap_usd": "2299890068.4280425642158096",
          "max_supply": null,
          "name": "EOS",
          "percent_change_24_hr": "6.8707627494412566",
          "price_usd": "2.5378234031415460",
          "rank": "4",
          "supply": "906245117.6000000000000000",
          "symbol": "EOS",
        },
        Object {
          "market_cap_usd": "2101330537.3210911914502786",
          "max_supply": null,
          "name": "Stellar",
          "percent_change_24_hr": "-0.5797575094501702",
          "price_usd": "0.1096095384438811",
          "rank": "5",
          "supply": "19171055431.4299000000000000",
          "symbol": "XLM",
        },
      ]
    }
    removeFromFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NewsContainer />,
        <FavoritesContainer
          abbrevCurrencies={
            Array [
              Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              },
              Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              },
              Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              },
              Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              },
              Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              },
              Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              },
              Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              },
              Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              },
              Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              },
              Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              },
            ]
          }
          addToFavorites={[MockFunction]}
          favorites={
            Array [
              Object {
                "market_cap_usd": "62506159630.7222615731538755",
                "max_supply": "21000000.0000000000000000",
                "name": "Bitcoin",
                "percent_change_24_hr": "0.8320854828625978",
                "price_usd": "3586.1585871814097615",
                "rank": "1",
                "supply": "17429837.0000000000000000",
                "symbol": "BTC",
              },
              Object {
                "market_cap_usd": "13612140064.7471135264064432",
                "max_supply": "100000000000.0000000000000000",
                "name": "XRP",
                "percent_change_24_hr": "3.9433058589033265",
                "price_usd": "0.3339389135808078",
                "rank": "2",
                "supply": "40762365544.0000000000000000",
                "symbol": "XRP",
              },
              Object {
                "market_cap_usd": "9875418262.8035156239002594",
                "max_supply": null,
                "name": "Ethereum",
                "percent_change_24_hr": "0.2048869680495625",
                "price_usd": "95.0699694086438060",
                "rank": "3",
                "supply": "103875264.9678000000000000",
                "symbol": "ETH",
              },
              Object {
                "market_cap_usd": "2299890068.4280425642158096",
                "max_supply": null,
                "name": "EOS",
                "percent_change_24_hr": "6.8707627494412566",
                "price_usd": "2.5378234031415460",
                "rank": "4",
                "supply": "906245117.6000000000000000",
                "symbol": "EOS",
              },
              Object {
                "market_cap_usd": "2101330537.3210911914502786",
                "max_supply": null,
                "name": "Stellar",
                "percent_change_24_hr": "-0.5797575094501702",
                "price_usd": "0.1096095384438811",
                "rank": "5",
                "supply": "19171055431.4299000000000000",
                "symbol": "XLM",
              },
            ]
          }
          removeFromFavorites={[MockFunction]}
        />,
        <LandingCurrencyContainer
          abbrevCurrencies={
            Array [
              Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              },
              Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              },
              Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              },
              Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              },
              Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              },
              Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              },
              Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              },
              Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              },
              Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              },
              Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              },
            ]
          }
        />,
      ],
      "className": "landing-literal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "abbrevCurrencies": Array [
            Object {
              "name": "Bitcoin",
              "percent_change": 0.83,
              "price": 3586.16,
              "rank": "1",
              "symbol": "BTC",
            },
            Object {
              "name": "XRP",
              "percent_change": 3.94,
              "price": 0.33,
              "rank": "2",
              "symbol": "XRP",
            },
            Object {
              "name": "Ethereum",
              "percent_change": 0.2,
              "price": 95.07,
              "rank": "3",
              "symbol": "ETH",
            },
            Object {
              "name": "EOS",
              "percent_change": 6.87,
              "price": 2.54,
              "rank": "4",
              "symbol": "EOS",
            },
            Object {
              "name": "Stellar",
              "percent_change": -0.58,
              "price": 0.11,
              "rank": "5",
              "symbol": "XLM",
            },
            Object {
              "name": "Tether",
              "percent_change": -0.37,
              "price": 1.02,
              "rank": "6",
              "symbol": "USDT",
            },
            Object {
              "name": "Bitcoin Cash",
              "percent_change": 12.32,
              "price": 100.65,
              "rank": "7",
              "symbol": "BCH",
            },
            Object {
              "name": "Litecoin",
              "percent_change": -1.97,
              "price": 28.99,
              "rank": "8",
              "symbol": "LTC",
            },
            Object {
              "name": "Bitcoin SV",
              "percent_change": -1.04,
              "price": 81.84,
              "rank": "9",
              "symbol": "BSV",
            },
            Object {
              "name": "TRON",
              "percent_change": 0.17,
              "price": 0.01,
              "rank": "10",
              "symbol": "TRX",
            },
          ],
          "addToFavorites": [MockFunction],
          "favorites": Array [
            Object {
              "market_cap_usd": "62506159630.7222615731538755",
              "max_supply": "21000000.0000000000000000",
              "name": "Bitcoin",
              "percent_change_24_hr": "0.8320854828625978",
              "price_usd": "3586.1585871814097615",
              "rank": "1",
              "supply": "17429837.0000000000000000",
              "symbol": "BTC",
            },
            Object {
              "market_cap_usd": "13612140064.7471135264064432",
              "max_supply": "100000000000.0000000000000000",
              "name": "XRP",
              "percent_change_24_hr": "3.9433058589033265",
              "price_usd": "0.3339389135808078",
              "rank": "2",
              "supply": "40762365544.0000000000000000",
              "symbol": "XRP",
            },
            Object {
              "market_cap_usd": "9875418262.8035156239002594",
              "max_supply": null,
              "name": "Ethereum",
              "percent_change_24_hr": "0.2048869680495625",
              "price_usd": "95.0699694086438060",
              "rank": "3",
              "supply": "103875264.9678000000000000",
              "symbol": "ETH",
            },
            Object {
              "market_cap_usd": "2299890068.4280425642158096",
              "max_supply": null,
              "name": "EOS",
              "percent_change_24_hr": "6.8707627494412566",
              "price_usd": "2.5378234031415460",
              "rank": "4",
              "supply": "906245117.6000000000000000",
              "symbol": "EOS",
            },
            Object {
              "market_cap_usd": "2101330537.3210911914502786",
              "max_supply": null,
              "name": "Stellar",
              "percent_change_24_hr": "-0.5797575094501702",
              "price_usd": "0.1096095384438811",
              "rank": "5",
              "supply": "19171055431.4299000000000000",
              "symbol": "XLM",
            },
          ],
          "removeFromFavorites": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "abbrevCurrencies": Array [
            Object {
              "name": "Bitcoin",
              "percent_change": 0.83,
              "price": 3586.16,
              "rank": "1",
              "symbol": "BTC",
            },
            Object {
              "name": "XRP",
              "percent_change": 3.94,
              "price": 0.33,
              "rank": "2",
              "symbol": "XRP",
            },
            Object {
              "name": "Ethereum",
              "percent_change": 0.2,
              "price": 95.07,
              "rank": "3",
              "symbol": "ETH",
            },
            Object {
              "name": "EOS",
              "percent_change": 6.87,
              "price": 2.54,
              "rank": "4",
              "symbol": "EOS",
            },
            Object {
              "name": "Stellar",
              "percent_change": -0.58,
              "price": 0.11,
              "rank": "5",
              "symbol": "XLM",
            },
            Object {
              "name": "Tether",
              "percent_change": -0.37,
              "price": 1.02,
              "rank": "6",
              "symbol": "USDT",
            },
            Object {
              "name": "Bitcoin Cash",
              "percent_change": 12.32,
              "price": 100.65,
              "rank": "7",
              "symbol": "BCH",
            },
            Object {
              "name": "Litecoin",
              "percent_change": -1.97,
              "price": 28.99,
              "rank": "8",
              "symbol": "LTC",
            },
            Object {
              "name": "Bitcoin SV",
              "percent_change": -1.04,
              "price": 81.84,
              "rank": "9",
              "symbol": "BSV",
            },
            Object {
              "name": "TRON",
              "percent_change": 0.17,
              "price": 0.01,
              "rank": "10",
              "symbol": "TRX",
            },
          ],
          "setFilter": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NewsContainer />,
          <FavoritesContainer
            abbrevCurrencies={
              Array [
                Object {
                  "name": "Bitcoin",
                  "percent_change": 0.83,
                  "price": 3586.16,
                  "rank": "1",
                  "symbol": "BTC",
                },
                Object {
                  "name": "XRP",
                  "percent_change": 3.94,
                  "price": 0.33,
                  "rank": "2",
                  "symbol": "XRP",
                },
                Object {
                  "name": "Ethereum",
                  "percent_change": 0.2,
                  "price": 95.07,
                  "rank": "3",
                  "symbol": "ETH",
                },
                Object {
                  "name": "EOS",
                  "percent_change": 6.87,
                  "price": 2.54,
                  "rank": "4",
                  "symbol": "EOS",
                },
                Object {
                  "name": "Stellar",
                  "percent_change": -0.58,
                  "price": 0.11,
                  "rank": "5",
                  "symbol": "XLM",
                },
                Object {
                  "name": "Tether",
                  "percent_change": -0.37,
                  "price": 1.02,
                  "rank": "6",
                  "symbol": "USDT",
                },
                Object {
                  "name": "Bitcoin Cash",
                  "percent_change": 12.32,
                  "price": 100.65,
                  "rank": "7",
                  "symbol": "BCH",
                },
                Object {
                  "name": "Litecoin",
                  "percent_change": -1.97,
                  "price": 28.99,
                  "rank": "8",
                  "symbol": "LTC",
                },
                Object {
                  "name": "Bitcoin SV",
                  "percent_change": -1.04,
                  "price": 81.84,
                  "rank": "9",
                  "symbol": "BSV",
                },
                Object {
                  "name": "TRON",
                  "percent_change": 0.17,
                  "price": 0.01,
                  "rank": "10",
                  "symbol": "TRX",
                },
              ]
            }
            addToFavorites={[MockFunction]}
            favorites={
              Array [
                Object {
                  "market_cap_usd": "62506159630.7222615731538755",
                  "max_supply": "21000000.0000000000000000",
                  "name": "Bitcoin",
                  "percent_change_24_hr": "0.8320854828625978",
                  "price_usd": "3586.1585871814097615",
                  "rank": "1",
                  "supply": "17429837.0000000000000000",
                  "symbol": "BTC",
                },
                Object {
                  "market_cap_usd": "13612140064.7471135264064432",
                  "max_supply": "100000000000.0000000000000000",
                  "name": "XRP",
                  "percent_change_24_hr": "3.9433058589033265",
                  "price_usd": "0.3339389135808078",
                  "rank": "2",
                  "supply": "40762365544.0000000000000000",
                  "symbol": "XRP",
                },
                Object {
                  "market_cap_usd": "9875418262.8035156239002594",
                  "max_supply": null,
                  "name": "Ethereum",
                  "percent_change_24_hr": "0.2048869680495625",
                  "price_usd": "95.0699694086438060",
                  "rank": "3",
                  "supply": "103875264.9678000000000000",
                  "symbol": "ETH",
                },
                Object {
                  "market_cap_usd": "2299890068.4280425642158096",
                  "max_supply": null,
                  "name": "EOS",
                  "percent_change_24_hr": "6.8707627494412566",
                  "price_usd": "2.5378234031415460",
                  "rank": "4",
                  "supply": "906245117.6000000000000000",
                  "symbol": "EOS",
                },
                Object {
                  "market_cap_usd": "2101330537.3210911914502786",
                  "max_supply": null,
                  "name": "Stellar",
                  "percent_change_24_hr": "-0.5797575094501702",
                  "price_usd": "0.1096095384438811",
                  "rank": "5",
                  "supply": "19171055431.4299000000000000",
                  "symbol": "XLM",
                },
              ]
            }
            removeFromFavorites={[MockFunction]}
          />,
          <LandingCurrencyContainer
            abbrevCurrencies={
              Array [
                Object {
                  "name": "Bitcoin",
                  "percent_change": 0.83,
                  "price": 3586.16,
                  "rank": "1",
                  "symbol": "BTC",
                },
                Object {
                  "name": "XRP",
                  "percent_change": 3.94,
                  "price": 0.33,
                  "rank": "2",
                  "symbol": "XRP",
                },
                Object {
                  "name": "Ethereum",
                  "percent_change": 0.2,
                  "price": 95.07,
                  "rank": "3",
                  "symbol": "ETH",
                },
                Object {
                  "name": "EOS",
                  "percent_change": 6.87,
                  "price": 2.54,
                  "rank": "4",
                  "symbol": "EOS",
                },
                Object {
                  "name": "Stellar",
                  "percent_change": -0.58,
                  "price": 0.11,
                  "rank": "5",
                  "symbol": "XLM",
                },
                Object {
                  "name": "Tether",
                  "percent_change": -0.37,
                  "price": 1.02,
                  "rank": "6",
                  "symbol": "USDT",
                },
                Object {
                  "name": "Bitcoin Cash",
                  "percent_change": 12.32,
                  "price": 100.65,
                  "rank": "7",
                  "symbol": "BCH",
                },
                Object {
                  "name": "Litecoin",
                  "percent_change": -1.97,
                  "price": 28.99,
                  "rank": "8",
                  "symbol": "LTC",
                },
                Object {
                  "name": "Bitcoin SV",
                  "percent_change": -1.04,
                  "price": 81.84,
                  "rank": "9",
                  "symbol": "BSV",
                },
                Object {
                  "name": "TRON",
                  "percent_change": 0.17,
                  "price": 0.01,
                  "rank": "10",
                  "symbol": "TRX",
                },
              ]
            }
          />,
        ],
        "className": "landing-literal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "abbrevCurrencies": Array [
              Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              },
              Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              },
              Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              },
              Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              },
              Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              },
              Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              },
              Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              },
              Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              },
              Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              },
              Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              },
            ],
            "addToFavorites": [MockFunction],
            "favorites": Array [
              Object {
                "market_cap_usd": "62506159630.7222615731538755",
                "max_supply": "21000000.0000000000000000",
                "name": "Bitcoin",
                "percent_change_24_hr": "0.8320854828625978",
                "price_usd": "3586.1585871814097615",
                "rank": "1",
                "supply": "17429837.0000000000000000",
                "symbol": "BTC",
              },
              Object {
                "market_cap_usd": "13612140064.7471135264064432",
                "max_supply": "100000000000.0000000000000000",
                "name": "XRP",
                "percent_change_24_hr": "3.9433058589033265",
                "price_usd": "0.3339389135808078",
                "rank": "2",
                "supply": "40762365544.0000000000000000",
                "symbol": "XRP",
              },
              Object {
                "market_cap_usd": "9875418262.8035156239002594",
                "max_supply": null,
                "name": "Ethereum",
                "percent_change_24_hr": "0.2048869680495625",
                "price_usd": "95.0699694086438060",
                "rank": "3",
                "supply": "103875264.9678000000000000",
                "symbol": "ETH",
              },
              Object {
                "market_cap_usd": "2299890068.4280425642158096",
                "max_supply": null,
                "name": "EOS",
                "percent_change_24_hr": "6.8707627494412566",
                "price_usd": "2.5378234031415460",
                "rank": "4",
                "supply": "906245117.6000000000000000",
                "symbol": "EOS",
              },
              Object {
                "market_cap_usd": "2101330537.3210911914502786",
                "max_supply": null,
                "name": "Stellar",
                "percent_change_24_hr": "-0.5797575094501702",
                "price_usd": "0.1096095384438811",
                "rank": "5",
                "supply": "19171055431.4299000000000000",
                "symbol": "XLM",
              },
            ],
            "removeFromFavorites": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "abbrevCurrencies": Array [
              Object {
                "name": "Bitcoin",
                "percent_change": 0.83,
                "price": 3586.16,
                "rank": "1",
                "symbol": "BTC",
              },
              Object {
                "name": "XRP",
                "percent_change": 3.94,
                "price": 0.33,
                "rank": "2",
                "symbol": "XRP",
              },
              Object {
                "name": "Ethereum",
                "percent_change": 0.2,
                "price": 95.07,
                "rank": "3",
                "symbol": "ETH",
              },
              Object {
                "name": "EOS",
                "percent_change": 6.87,
                "price": 2.54,
                "rank": "4",
                "symbol": "EOS",
              },
              Object {
                "name": "Stellar",
                "percent_change": -0.58,
                "price": 0.11,
                "rank": "5",
                "symbol": "XLM",
              },
              Object {
                "name": "Tether",
                "percent_change": -0.37,
                "price": 1.02,
                "rank": "6",
                "symbol": "USDT",
              },
              Object {
                "name": "Bitcoin Cash",
                "percent_change": 12.32,
                "price": 100.65,
                "rank": "7",
                "symbol": "BCH",
              },
              Object {
                "name": "Litecoin",
                "percent_change": -1.97,
                "price": 28.99,
                "rank": "8",
                "symbol": "LTC",
              },
              Object {
                "name": "Bitcoin SV",
                "percent_change": -1.04,
                "price": 81.84,
                "rank": "9",
                "symbol": "BSV",
              },
              Object {
                "name": "TRON",
                "percent_change": 0.17,
                "price": 0.01,
                "rank": "10",
                "symbol": "TRX",
              },
            ],
            "setFilter": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
